### PR TITLE
Add method to handle builds added messages

### PR DESCRIPTION
**Pull request by sarah256**

### CODE REVIEW

File: estuary_updater/handlers/errata.py [hunk 1]
```diff
@@ -67,3 +89,32 @@ def handle(self, msg):
         for bug in bugs:
             bug = BugzillaBug.get_or_create({'id_': bug['bug']['id']})[0]
             advisory.attached_bugs.connect(bug)
+
+    def builds_added_handler(self, msg):
```
[the remark this answers is not in the document]
Comment (sarah256) [reply estuary_updater/handlers/errata.py]: @mprahl will do this in a separate PR

File: estuary_updater/handlers/errata.py [hunk 2]
```diff
@@ -67,3 +89,37 @@ def handle(self, msg):
...
+        :param dict msg: a message to be processed
+        """
+        self.koji_session = koji.ClientSession(self.config['estuary_updater.koji_url'])
+
+        nvr = msg['body']['headers']['brew_build']
+
+        try:
+            koji_build = self.koji_session.getBuild(nvr, strict=True)
+        except Exception:
+            log.warning('Failed to get brew build with NVR {0}'.format(nvr))
```
Comment (mprahl) [inline estuary_updater/handlers/errata.py]: Can you make this `log.error` now that you are raising the exception again

File: tests/handlers/test_errata.py [hunk 3]
```diff
@@ -46,3 +47,42 @@ def test_errata():
...
+    with open(path.join(message_dir, 'errata', 'builds_added.json'), 'r') as f:
+        msg = json.load(f)
+    # Make sure the handler can handle the message
+    assert ErrataHandler.can_handle(msg) is True
+    # Instantiate the handler
+    handler = ErrataHandler(config)
+    # Run the handler
+    handler.handle(msg)
+
+    assert advisory is not None
```
Comment (mprahl) [inline tests/handlers/test_errata.py]: This line can be removed.

File: tests/handlers/test_errata.py [hunk 4]
```diff
@@ -46,3 +47,42 @@ def test_errata():
...
+
+    build = KojiBuild.nodes.get_or_none(id_=123456)
+    assert build is not None
+    assert build.name == 'openstack-zaqar-container'
+    assert build.version == '13.0'
+    assert build.release == '45'
```
Comment (mprahl) [inline tests/handlers/test_errata.py]: It'd be nice to test that the other properties were set.

File: estuary_updater/handlers/errata.py [hunk 5]
```diff
@@ -67,3 +90,44 @@ def handle(self, msg):
...
+
+        nvr = msg['body']['headers']['brew_build']
+
+        try:
+            koji_build = self.koji_session.getBuild(nvr, strict=True)
+        except Exception:
+            log.error('Failed to get brew build with NVR {0}'.format(nvr))
+            raise
+
+        build_params = {
```
Comment (mprahl) [inline estuary_updater/handlers/errata.py]: Is `start_time` a value that's not available?

Comment (mprahl) [inline estuary_updater/handlers/errata.py]: Is `extra` a value that's not available?

File: estuary_updater/handlers/errata.py [hunk 6]
```diff
@@ -67,3 +90,44 @@ def handle(self, msg):
...
+            'release': koji_build['release'],
+            'state': koji_build['state'],
+            'version': koji_build['version']
+        }
+
+        advisory = Advisory.get_or_create({
+            'id_': msg['body']['headers']['errata_id']
+        })[0]
+
+        build = KojiBuild.create_or_update(build_params)[0]
```
Comment (mprahl) [inline estuary_updater/handlers/errata.py]: Optional: For readability, I'd prefer if the creation of `build` was right after `build_params` was defined. That way you can see what `build_params` is used for right away.

File: estuary_updater/handlers/errata.py [hunk 7]
```diff
@@ -67,3 +90,44 @@ def handle(self, msg):
...
+
+        advisory.attached_builds.connect(build)
+
+        owner = User.get_or_create({
+            'username': koji_build['owner_name'] + '@redhat.com'
```
Comment (mprahl) [inline estuary_updater/handlers/errata.py]: This should be:
```python
owner = User.create_or_update({
    'username': koji_build['owner_name'],
    'email': '{0}@redhat.com'.format(koji_build['owner_name'])
})
```

File: tests/handlers/test_errata.py [hunk 8]
```diff
@@ -46,3 +48,51 @@ def test_errata():
...
+    # Make sure the handler can handle the message
+    assert ErrataHandler.can_handle(msg) is True
+    # Instantiate the handler
+    handler = ErrataHandler(config)
+    # Run the handler
+    handler.handle(msg)
+
+    build = KojiBuild.nodes.get_or_none(id_=123456)
+    owner = User.nodes.get_or_none(username='emusk')
+    assert build is not None
```
Comment (mprahl) [inline tests/handlers/test_errata.py]: Could you please test that `start_time` was set as well?

**Comment by mprahl** [comment]
:+1: with one minor comment. After it's addressed, feel free to merge it!

**Pull request merged**